### ISSUE TITLE
Accessing ifc file from entity_instance

### ISSUE THREAD

**Issue by Andrej730** [IfcOpenShell/IfcOpenShell#4220]
Just noticed that there seems to be `file` attribute in `entity_instance` and it appears in `dir` but it's seems to be unaccessible. 
Is it possible to get the file object from the entity?

```python
ifc_model = ifcopenshell.file()
wall = ifc_model.createIfcWall()
print("file" in dir(wall)) # True
print(wall.file) # AttributeError: entity instance of type 'IFC4.IfcWall' has no attribute 'file'
```

Update. Found that it is accessible from `wall.wrapped_data.file`. A bit odd that it occurs in `dir(wall)` though.

**Comment by Andrej730** [comment]
> With the above line activated and my fix, does it work for you?

Yes, it works.

**Comment by aothms** [comment]
> Update. Found that it is accessible from wall.wrapped_data.file. A bit odd that it occurs in dir(wall) though.

yes, but it's really funky, because not in all cases that will be populated. So they serve different purposes actually.

*wall.wrapped_data.file* prevent garbage collection of file when instances of file are still available

*wall.file* lookup file from map based on C++ pointer address, works even in cases where the python instance is not created by us, such as using ifcopenshell.geom.tree

---

Btw, I just committed another fix, because obviously storing a ref in a map prevents the file from being freed, so I changed it to a weakref.
